apis/v1beta1/shared_types.go

```diff
@@ -275,6 +275,15 @@ const (
 	// This reason is used with the "ResolvedRefs" condition when one of the
 	// Route's rules has a reference to a resource that does not exist.
 	RouteReasonBackendNotFound RouteConditionReason = "BackendNotFound"
+
+	// This condition indicates whether incompatible filters are present on
+	// a Route. Controllers should only add this condition with status true
+	// when an incompatible combination is present.
+	RouteConditionIncompatibleFilters RouteConditionType = "IncompatibleFilters"
```

Comment:
If we decide this does require a new condition, I think we'd need to have the full structure we usually attach to a condition definition:

https://github.com/kubernetes-sigs/gateway-api/blob/5fdab20c7a591ad914b45a2d62e855f167192278/apis/v1beta1/shared_types.go#L193-L212